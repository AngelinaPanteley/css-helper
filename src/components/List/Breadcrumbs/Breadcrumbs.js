import React, { PureComponent } from 'react';

import styles from './Breadcrumbs.scss';

class Breadcrumbs extends PureComponent {
  render() {
    const { pageNumber, pageAmount } = this.props;
    const optionArray = [];
    for (let i = 0; i < pageAmount; ++i) {
      optionArray.push(<option key={i + 1} value={i + 1}>{i + 1}</option>);
    }

    return (
      <div>
        <button
          onClick={() => { this.props.onChange(pageNumber - 1) }}
          disabled={pageNumber - 1 === 0}>
          Prev
        </button>
        <select value={pageNumber}
          onChange={(e) => { this.props.onChange(+e.target.value) }}>
          {optionArray}
        </select>
        <button
          onClick={() => { this.props.onChange(pageNumber + 1) }}
          disabled={pageNumber + 1 > pageAmount}>
          Next
          </button>
      </div>
    )
  }
}

export default Breadcrumbs;