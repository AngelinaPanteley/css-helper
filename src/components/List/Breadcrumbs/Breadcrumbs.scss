@import '../../../variables';

.Link {
  background: none;
  border: none;
  color: $color-white;
  outline: none;
  cursor: pointer;
  font-size: 17px;
  transition: $transition;

  &:hover {
    text-shadow: 0 0 1px $color-white;
    transition: $transition;
  }
}

.Wrapper {
  display: flex;
  justify-content: center;
  margin: 20px 0;
}