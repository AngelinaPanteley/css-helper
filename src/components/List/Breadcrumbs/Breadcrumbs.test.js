"use strict";

import React from 'react';
import renderer from 'react-test-renderer';
import { expect as chaiExpect } from 'chai';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16.1';
import Breadcrumbs from './Breadcrumbs.js';

configure({ adapter: new Adapter() });

describe('>>>Breadcrumbs', () => {

  //SNAPSHOTS

  const props = {
    onChange: () => { },
  };

  const breadcrumbsComponent = <Breadcrumbs {...props} />;
  const component = renderer.create(breadcrumbsComponent);

  it('renders correctly', () => {
    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();
  });

  //SHALLOW RENDERING

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(breadcrumbsComponent);
  });

  it('renders prev and next buttons', () => {
    chaiExpect(wrapper.find('button')).to.have.lengthOf(2);
  });

  it('renders select of page number', () => {
    chaiExpect(wrapper.find('select')).to.have.lengthOf(1);
  });
});
