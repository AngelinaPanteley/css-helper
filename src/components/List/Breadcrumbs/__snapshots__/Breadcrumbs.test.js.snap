// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>Breadcrumbs renders correctly 1`] = `
<div>
  <button
    disabled={false}
    onClick={[Function]}
  >
    &lt;   Prev
  </button>
  <select
    onChange={[Function]}
  />
  <button
    disabled={false}
    onClick={[Function]}
  >
    Next   &gt;
  </button>
</div>
`;
